src/master/manager.cpp

```diff
@@ -641,17 +643,59 @@ class ManagerProcess : public ProtobufProcess<ManagerProcess>
 
       LOG(INFO) << "Initializing the replicated log.";
 
-      if (masterConfig.has_zk()) {
-        LOG(INFO) << "Using replicated log with zookeeper for leader"
-                  << " election.";
-        Try<zookeeper::URL> url =
-          zookeeper::URL::parse(masterConfig.zk().url());
+      // The ZK URL and Quorum can be specified through environment
+      // variables or through the JSON config.
+      Option<string> zkURL = os::getenv(MESOS_ZK);
+      if (zkURL.isNone()) {
+        // Check the JSON only if ZK URL has not been specified in the
+        // environment variables.
+        if (masterConfig.has_zk()) {
+           zkURL = masterConfig.zk().url();
```

Comment:
BUG: Now `url` is optional too, so you need to check if `masterConfig.zk().has_url()` too